modules/adqueryBidAdapter.js

```diff
@@ -0,0 +1,215 @@
+import {registerBidder} from '../src/adapters/bidderFactory.js';
+import {BANNER} from '../src/mediaTypes.js';
+import * as utils from '../src/utils.js';
+import { getStorageManager } from '../src/storageManager.js';
+export const ADAPTER_VERSION = 1;
+
+const ADQUERY_GVLID = 902;
+const ADQUERY_BIDDER_CODE = 'adquery';
+const ADQUERY_BIDDER_DOMAIN_PROTOCOL = 'https';
+const ADQUERY_BIDDER_DOMAIN = 'bidder.adquery.io';
+const ADQUERY_USER_SYNC_DOMAIN = ADQUERY_BIDDER_DOMAIN_PROTOCOL + '://' + ADQUERY_BIDDER_DOMAIN + '/prebid/userSync?1=1';
+const ADQUERY_LOG_PREFIX = 'Adquery: ';
+const ADQUERY_DEFAULT_CURRENCY = 'PLN';
+const ADQUERY_NET_REVENUE = true;
+const ADQUERY_TTL = 360;
+const storage = getStorageManager(ADQUERY_GVLID);
+/** @type {BidderSpec} */
+export const spec = {
+  code: ADQUERY_BIDDER_CODE,
+  gvlid: ADQUERY_GVLID,
+  supportedMediaTypes: [BANNER],
+
+  /** f
+   * @param {object} bid
+   * @return {boolean}
+   */
+  isBidRequestValid: (bid) => {
+    if (!(bid && bid.params && (bid.params.placementId))) {
+      return false;
+    }
+    return true;
+  },
+
+  /**
+   * @param {BidRequest[]} bidRequests
+   * @param {*} bidderRequest
+   * @return {ServerRequest}
+   */
+  buildRequests: (bidRequests, bidderRequest) => {
+    const ua = navigator.userAgent;
+    let qid = Math.random().toString(36).substring(2) + Date.now().toString(36);
+    if (storage.getDataFromLocalStorage('qid')) {
+      qid = storage.getDataFromLocalStorage('qid');
+    } else {
+      storage.setDataInLocalStorage('qid', qid);
+    }
+    let bid = bidRequests[0];
```

Comment:
Why bidding only on 1st adUnit?  
If user requests bids for multiple adUnits at the same time, your bid adapter will send only bid request for 1 adUnit.  

If your backend, at the moment, doesn't support multi bid then in this function you should return array of bid request objects, not just single object. Then pbjs will send separate http request to your backed for each adUnit.

Reply:
I fixed